Code/Framework/AzToolsFramework/AzToolsFramework/Script/LuaEditorSystemComponent.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Script/LuaEditorSystemComponent.h>
+
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/IO/SystemFile.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/string/conversions.h>
+#include <AzFramework/StringFunc/StringFunc.h>
+#include <AzToolsFramework/API/ToolsApplicationAPI.h>
+#include <QIcon>
+
+namespace AzToolsFramework
+{
+    namespace Script
+    {
+        void LuaEditorSystemComponent::Reflect(AZ::ReflectContext* context)
+        {
+            AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context);
+            if (serialize)
+            {
+                serialize->Class<LuaEditorSystemComponent, AZ::Component>();
+            }
+        }
+
+        void LuaEditorSystemComponent::Activate()
+        {
+            AzToolsFramework::AssetBrowser::AssetBrowserInteractionNotificationBus::Handler::BusConnect();
+            AzToolsFramework::AssetBrowser::AssetBrowserFileCreationNotificationBus::Handler::BusConnect(LuaComponentScriptBusId);
+        }
+
+        void LuaEditorSystemComponent::Deactivate()
+        {
+            AzToolsFramework::AssetBrowser::AssetBrowserInteractionNotificationBus::Handler::BusDisconnect();
```

Comment:
Suggestion: it's advised that buses should be disconnected in reverse order of connection.

Reply:
Good call, thank you. Done in the next commit.